src/renderer.ts

```diff
@@ -204,10 +211,21 @@ class Renderer extends EventEmitter<RendererEvents> {
     return [div, shadow]
   }
 
+  /** Wavesurfer itself calls this method. Do not call it manually. */
   setOptions(options: WaveSurferOptions) {
+    if (this.options.container !== options.container) {
+      const newParent = this.parentFromOptionsContainer(options.container)
+      newParent.appendChild(this.container)
+
+      this.parent = newParent
+    }
+
     this.options = options
-    // Re-render the waveform
-    this.reRender()
+
+    // Re-render the waveform if render related options have been changed
+    if (!options.container || Object.keys(options).length > 1) {
```

Comment:
The problem was caused by `Object.keys(options).length` returning 17 instead of 1 as we earlier expected. 

In the `wavesurfer.ts` file the `setOptions` method does an `Object.assing(...)` before it passes the value to the renderer.
Do i just add the equals.ts back in and if so where would you like the logic to reside in?

Reply:
Good point. Then I would just remove this if and let it re-render every time. It would be actially good because then it will also have a chance to adjust the width to that of the new container.

I understand your particular use case was aiming to avoid re-renderings, and that was the whole point. I would suggest for those use cases to move (appendChild) the entire outer container and not the wavesurfer container.

Reply:
Tests now pass. Im continuing with the way you are suggesting therefore i don't really need this feature anymore.

It is up to you if you want to merge it or not.